package.json

```diff
@@ -24,13 +24,13 @@
     "semantic-release": "semantic-release"
   },
   "dependencies": {
-    "@typescript-eslint/utils": "^5.57.0",
+    "@typescript-eslint/utils": "^6.0.0",
```

Comment:
Btw I just realized that since this `@typescript-eslint/utils` dependency is direct it is not going to be possible to use this package with eslint v7 as ppl will require older `@typescript-eslint/utils` v5 just like in our BC test case.
So instead we should move it to peer dependency and add a version range that includes v5 and v6 just like with typescript and eslint.

Reply:
Well, `@typescript-eslint/utils`at least declares support for eslint 7 still:
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/utils/package.json
https://github.com/typescript-eslint/typescript-eslint/blob/2d66716de61702f7711273612c65ed2ad153c0e7/packages/eslint-plugin/package.json#L95
https://github.com/typescript-eslint/typescript-eslint/blob/2d66716de61702f7711273612c65ed2ad153c0e7/packages/parser/package.json#L51

It seems to just be the newly extracted rule-tester package that has a hard dependency on 8+

But it does seem like it's making it hard to test eslint 7 properly, and I guess moving it to a peerDependency can be a good idea anyway. I'll try to make it work and actually run tests with those versions locally to verify!

Reply:
I can still do one more test at moving to a peerDependency if you want, now that I at least figured out why the tests didn't run properly on eslint 7. But I think this was the big issue with the tests there.

Reply:
Well since they are supporting eslint v7 I guess we can keep it in dependencies for now.